parlai/tasks/squad/agents.py

```diff
@@ -383,15 +304,40 @@ class FulldocsentenceTeacher(FulldocTeacher):
     label candidates, and the label as the sentence containing the answer.
 
     Some punctuation may be removed for tokenization purposes.
+
+    If `include_context` is False, the teacher returns action dict in the
+    following format:
+    {
+        'context': <context>,
+        'text': <question>,
+        'labels': <sentences containing the true answer>,
+        'label_candidates': <all sentences in the context>,
+        'episode_done': True,
+        'answer_starts': <index of start of answer in context>
+    }
+    Otherwise, the 'text' field contains <context>\n<question> and there is
+    no separate context field.
     """
     def __init__(self, opt, shared=None):
         super().__init__(opt, shared)
         self.sent_tok = get_sentence_tokenizer()
+        self.include_context = opt.get('include_context', False)
+
+    @staticmethod
+    def add_cmdline_args(argparser):
+        agent = argparser.add_argument_group(
+            'SQuAD Fulldoc Sentence Teacher Arguments'
+        )
+        agent.add_argument('--include-context',  type='bool', default=False,
+                           help='include context within text instead of as a '
+                           'separate field')
 
     def get(self, episode_idx, entry_idx=None):
+        action = {}
         episode = self.episodes[episode_idx][entry_idx]
-        entry = {'episode_done': episode['episode_done']}
-        context = ' '.join(episode['text'].split('\n')[:-1]).replace('\xa0', ' ')
+        context = ' '.join(
+            episode['text'].split('\n')[:-1]
+        ).replace('\xa0', ' ')
```

Comment:
what's up with the weird character replacement?

Reply:
great question...this is a non breaking space character and they were left in the text. in some cases, the correct answer span in the text contains this character and so if we don't remove it, the model has no possibility of getting the answer correct (if it returns an answer span)

Reply:
Makes sense. It'd be nice if we had a comment here briefly explaining that.

Reply:
added with commit ff68e3